README.md

```diff
@@ -348,6 +349,72 @@ raw JSON.
 
 `case..of` is clever as it will generate more efficent JS, and it also allows you to catch unmatched patterns at compile time. It's also cheap to extend this data with something more useful later on, like if you need to add another branch. This saves code diffs.
 
+## Identifiers
+
+### Prefer the use of union types over simple types for identifiers
+
+Using a type alias for a unique identifier allows for easy mistakes supplying any old string or integer that happens to be in scope, rather than an actual identifier. Prefer using a single case union instead:
+
+```elm
+-- Don't do this --
+type alias Student =
+    { id : StudentId
+    , name : String
+    , age : Int
+    }
+
+
+type alias StudentId =
+    String
+
+getStudentDetails : StudentId -> Student
+getStudentDetails sid =
+    -- Passing a name in here by mistake will compile fine --
+    ...
+```
+
+```elm
+-- Do this instead --
+type alias Student =
+    { id : StudentId
+    , name : String
```

Comment:
Should we in a similar fashion encourage this to be a `StudentName`, and `age : `Age`?

Reply:
That is a very good (although maybe orthogonal?) question. I mean, the broader thinking here is that anything which has a domain specific meaning should have a domain specific type: but the flip side is that StudentName has very few constraints and you're likely to do a lot of string operations on it.

If you can think of a reasonably clear set of guidelines for where to switch to a type, I'm all for it!

Reply:
One guideline is "might the underlying implementation change?"

For example, we might conceivably switch from integers to strings for database identifiers. (It's very unlikely, but conceivable nonetheless.)

`name` is always going to be a string though.